lib/events/onPush.ts

```diff
@@ -335,81 +303,54 @@ ${captureLog.log.trim()}
 	},
 };
 
-export interface Annotation {
-	path: string;
-	line: number;
-	column: number;
-	severity: "failure" | "notice" | "warning";
-	title: string;
-	message: string;
-}
-
-function extractAnnotations(lines: string): Annotation[] {
-	const logs = lines.split("\n");
-	const annotations = [];
-	for (const matcher of Matchers) {
-		for (const pattern of matcher.pattern) {
-			for (const l of logs) {
-				const match = new RegExp(pattern.regexp, "g").exec(l.trim());
-				if (match) {
-					annotations.push({
-						match: match[0],
-						path: match[pattern.groups.path],
-						line: match[pattern.groups.line],
-						column: match[pattern.groups.column],
-						severity: mapSeverity(
-							(
-								match[pattern.groups.severity] || "error"
-							).toLowerCase(),
-						),
-						message: match[pattern.groups.message],
-						title: match[pattern.groups.title],
-					});
-				}
-			}
-		}
-	}
-	return annotations;
-}
-
-function mapSeverity(severity: string): "notice" | "warning" | "failure" {
-	switch (severity.toLowerCase()) {
-		case "error":
-			return "failure";
-		case "warning":
-		case "warn":
-			return "warning";
-		case "info":
-		case "information":
-			return "notice";
-		default:
-			return "notice";
-	}
-}
-
 const NpmVersionStep: NpmStep = {
 	name: "version",
-	runWhen: async ctx => ctx.configuration?.[0]?.parameters.publish,
+	runWhen: async ctx =>
+		ctx.configuration?.[0]?.parameters.publish &&
+		ctx.configuration?.[0]?.parameters.publish !== "no",
 	run: async (ctx, params) => {
 		const push = ctx.data.Push[0];
 		let pj = await fs.readJson(params.project.path("package.json"));
 
-		let pjVersion = pj.version;
-		if (!pjVersion || pjVersion.length === 0) {
-			pjVersion = "0.0.0";
-		}
+		const pjVersion =
+			pj.version ||
+			(await github.nextTag(params.project.id, "patch")) ||
+			"0.1.0";
+
+		const repo = push.repo;
+		const credential = await ctx.credential.resolve(
+			secret.gitHubAppToken({
+				owner: repo.owner,
+				repo: repo.name,
+				apiUrl: repo.org.provider.apiUrl,
+			}),
+		);
+		const octokit = new Octokit({
```

Comment:
I'd prefer that to use `github.api()` from `@atomist/skill` to not introduce a dependency to octokit in this project's `package.json`. Additionally the Octokit instance from skill is configured with some plugins out of the box.

Reply:
I did not know that existed. I'll check it out.

Reply:
Does `github.api()` handle all the auth for you?

Reply:
Looks like not.